auth/auth.go

```diff
@@ -101,6 +107,33 @@ func (a ACL) VerifyPeerCertificateServer(rawCerts [][]byte, verifiedChains [][]*
 		return nil
 	}
 
+	input := map[string]interface{}{
+		"certificate": cert,
```

Comment:
What does the input into the policy look like if we provide the cert like this? Any pre-processing we should do in order to make it easier to write a policy against? I'm wondering what extensions would look like in the input, for example.

Reply:
Yep, not sure what happens with extensions. For basic stuff, it is just the x509 go object reflected. eg: `input.certificate.Subject.CommonName`

Will check for extensions.

I would have a slight preference for *not* doing any pre-processing / forcing into a custom API - and instead just letting people access the certificate and do whatever they want.

By the way, I am wondering if we should also expose other connection information to the policy, besides the certificate (client ip?)?

Reply:
I prefer less processing too, but at least we should make sure we have DNS and URI SANs easily available because that's how modern standard certificates are issued for browsers. Exposing other info sounds useful, esp. the IP address of the client could be interesting.

Reply:
Agreed on making sure we can access these  - will check.

About the IP - after checking quickly: might be tricky to get the IP - since we do not have access to the connection inside `VerifyPeerCertificateClient` - any suggestion on how to best pass it along here?